core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -76,18 +80,23 @@ class HistoryServer(
         return
       }
 
-      val appId = parts(1)
+      val appKey =
+        if (parts.length == 3) {
+          s"${parts(1)}/${parts(2)}"
```

Comment:
can you add a comment on what these parts represent? Maybe add an example in the comment?